pkg/dump/dump.go

```diff
@@ -142,3 +160,174 @@ func getTimescaleInfo(dbURI string) (util.TsInfo, error) {
+// compression job requests exclusive lock (waits on main dump process) 3) child dump
+// process requests read lock (waits on compression process). Essentially, any concurrent
+// DDL operation could cause this problem, but the user can put off DDL, but this is
+// harder with scheduled operations like compression jobs.
+
+// The simple solution would seem to be pausing jobs while we are doing the dump and
+// resuming them when we've finished. Unfortunately, a full pause is problematic as, if
+// the dump process were to crash, you'd end up with your jobs in a state where they would
+// never run. In order to avoid that, we instead create a process that checks for
+// problematic jobs scheduled to start in the next ten minutes and moves them into the
+// future by 15-20 minutes. This way, if the dump process crashes, the worst that can
+// happen is that your job is put off by 20 minutes. We also keep track of any jobs we
+// rescheduled and try to put them back on schedule when we finish the dump.
+func compressionJobMover(dbURI string, wg *sync.WaitGroup, jobsStopped chan bool, cleanup chan bool, pauseUDAs bool, verbose bool) {
+	defer wg.Done()
+
+	movedJobs := make(map[int64]time.Time)
+	runningJobs := true
+
+	ticker := time.NewTicker(30 * time.Second)
+	conn, err := util.GetDBConn(context.Background(), dbURI)
+	if err != nil {
+		compressionJobMoverWarn(err)
+		return
+	}
+	defer conn.Close(context.Background())
+	var tsMajorVersion int
+	err = conn.QueryRow(context.Background(), "SELECT split_part(extversion, '.', 1)::INT FROM pg_catalog.pg_extension WHERE extname='timescaledb' LIMIT 1").Scan(&tsMajorVersion)
+	if err != nil {
+		compressionJobMoverWarn(err)
+		return
+	}
+	runningJobsSQL, moveSQL, replaceJobSQL, err := compressionJobMoverSQL(tsMajorVersion, pauseUDAs)
+	if err != nil {
+		compressionJobMoverWarn(err)
+		return
+	}
+	var runningJobIDs string
+	for {
+		rows, err := conn.Query(context.Background(), moveSQL)
+		if err != nil {
+			compressionJobMoverWarn(err)
+			return
+		}
+		var (
+			jobID     int64
+			origStart time.Time
+		)
+		for rows.Next() {
+			if err = rows.Scan(&jobID, &origStart); err != nil {
+				compressionJobMoverWarn(err)
+				return
+			}
+			if _, ok := movedJobs[jobID]; !ok {
+				if verbose {
+					fmt.Printf("%sMoved Job: %d\n", time.Now().Format("2006/01/02 15:04:05 "), jobID)
+				}
+				movedJobs[jobID] = origStart
+			}
+		}
+		rows.Close()
+		if runningJobs {
+
+			err = conn.QueryRow(context.Background(), runningJobsSQL).Scan(&runningJobIDs)
+			if err != nil && err != pgx.ErrNoRows {
+				compressionJobMoverWarn(err)
+				return
+			}
+			if err == pgx.ErrNoRows {
+				err = nil
+				runningJobs = false
+				fmt.Printf("%sJobs: %s have stopped, continuing \n ", time.Now().Format("2006/01/02 15:04:05 "), runningJobIDs)
+				jobsStopped <- true // signal back to our parent that it's safe to move on, the jobs have stopped
```

Comment:
We should never be writting again so just close the channel
```suggestion
				close(jobsStopped)// signal back to our parent that it's safe to move on, the jobs have stopped
```